ext/node/ops/fs.rs

```diff
@@ -78,3 +79,140 @@ where
   Ok(())
 }
+
+#[derive(Debug, Serialize)]
+pub struct StatFs {
+  #[serde(rename = "type")]
+  pub typ: u64,
+  pub bsize: u64,
+  pub blocks: u64,
+  pub bfree: u64,
+  pub bavail: u64,
+  pub files: u64,
+  pub ffree: u64,
+}
+
+#[op2]
+#[serde]
+pub fn op_node_statfs<P>(
+  state: Rc<RefCell<OpState>>,
+  #[string] path: String,
+  bigint: bool,
+) -> Result<StatFs, AnyError>
+where
+  P: NodePermissions + 'static,
+{
+  {
+    let mut state = state.borrow_mut();
+    state
+      .borrow_mut::<P>()
+      .check_read_with_api_name(Path::new(&path), Some("node:fs.statfs"))?;
+    state
+      .borrow_mut::<P>()
+      .check_sys("statfs", "node:fs.statfs")?;
+  }
+  #[cfg(unix)]
+  {
+    use std::ffi::OsStr;
+    use std::os::unix::ffi::OsStrExt;
+
+    let path = OsStr::new(&path);
+    let mut cpath = path.as_bytes().to_vec();
+    cpath.push(0);
+    if bigint {
+      #[cfg(not(target_os = "macos"))]
+      // SAFETY: `cpath` is NUL-terminated and result is pointer to valid statfs memory.
+      let (code, result) = unsafe {
+        let mut result: libc::statfs64 = std::mem::zeroed();
+        (libc::statfs64(cpath.as_ptr() as _, &mut result), result)
+      };
+      #[cfg(target_os = "macos")]
+      // SAFETY: `cpath` is NUL-terminated and result is pointer to valid statfs memory.
+      let (code, result) = unsafe {
+        let mut result: libc::statfs = std::mem::zeroed();
+        (libc::statfs(cpath.as_ptr() as _, &mut result), result)
+      };
+      if code == -1 {
+        return Err(std::io::Error::last_os_error().into());
+      }
+      Ok(StatFs {
+        typ: result.f_type as _,
+        bsize: result.f_bsize as _,
+        blocks: result.f_blocks as _,
+        bfree: result.f_bfree as _,
+        bavail: result.f_bavail as _,
+        files: result.f_files as _,
+        ffree: result.f_ffree as _,
+      })
+    } else {
+      // SAFETY: `cpath` is NUL-terminated and result is pointer to valid statfs memory.
+      let (code, result) = unsafe {
+        let mut result: libc::statfs = std::mem::zeroed();
+        (libc::statfs(cpath.as_ptr() as _, &mut result), result)
+      };
+      if code == -1 {
+        return Err(std::io::Error::last_os_error().into());
+      }
+      Ok(StatFs {
+        typ: result.f_type as _,
+        bsize: result.f_bsize as _,
+        blocks: result.f_blocks as _,
+        bfree: result.f_bfree as _,
+        bavail: result.f_bavail as _,
+        files: result.f_files as _,
+        ffree: result.f_ffree as _,
+      })
+    }
+  }
+  #[cfg(windows)]
+  {
+    use deno_core::anyhow::anyhow;
+    use std::ffi::OsStr;
+    use std::os::windows::ffi::OsStrExt;
+    use windows_sys::Win32::Storage::FileSystem::GetDiskFreeSpaceW;
+
+    let _ = bigint;
+    // Using a vfs here doesn't make sense, it won't align with the windows API
+    // call below.
+    #[allow(clippy::disallowed_methods)]
+    let path = Path::new("Cargo.toml").canonicalize().unwrap();
```

Comment:
Wouldn't `canonicalize()` fail if `Cargo.toml` is not present?

Reply:
I'm not sure why we need this. The user `path` argument is shadowed by this hardcoded path to Cargo.toml

Reply:
I must have copy-pasted this from my windows playground repo... that's frustrating there was no lint error.